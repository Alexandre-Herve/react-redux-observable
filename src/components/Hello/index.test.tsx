import React from 'react';
import sinon from 'sinon';
import { shallow } from 'enzyme';
import { Hello } from './';

test('matches snapshot', () => {
  const onSetGreeting = sinon.spy();
  const greeting = 'Hi';
  const wrapper = shallow(<Hello greeting={greeting} onSetGreeting={onSetGreeting} />);
  expect(wrapper).toMatchSnapshot();
});

test('renders greeging', () => {
  const onSetGreeting = sinon.spy();
  const greeting = 'Hi';
  const wrapper = shallow(<Hello greeting={greeting} onSetGreeting={onSetGreeting} />);
  expect(wrapper.text()).toContain(greeting);
});
