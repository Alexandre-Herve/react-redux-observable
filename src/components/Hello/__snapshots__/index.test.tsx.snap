// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    greeting="Hi"
    onSetGreeting={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="greeting"
        >
          Hello lol 
          Hi
        </div>,
        <button
          onClick={[Function]}
        >
          tagada
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Hello lol ",
            "Hi",
          ],
          "className": "greeting",
        },
        "ref": null,
        "rendered": Array [
          "Hello lol ",
          "Hi",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "tagada",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "tagada",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="greeting"
          >
            Hello lol 
            Hi
          </div>,
          <button
            onClick={[Function]}
          >
            tagada
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Hello lol ",
              "Hi",
            ],
            "className": "greeting",
          },
          "ref": null,
          "rendered": Array [
            "Hello lol ",
            "Hi",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "tagada",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "tagada",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
